gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -108,6 +124,165 @@ URLs - this aspect is taken care of in the preferences server component.
+     *
+     * @hash (Object) The set we are assigning incrementing integers to.
+     * @item (Object) The next value to add to the set. If this item is already
+     *     in the set, the existing value for it will be unchanged.
+     * @returns Does not return a value, operates directly on the hash.
+     */
+    gpii.ontologyHandler.floydWarshall.incrementingHashAppend = function (hash, item) {
+        if (!hash[item]) {
+            hash[item] = Object.keys(hash).length + 1;
+        }
+    };
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshall.fullAlgorithmWithReconstruction
+     * to initialize the core matrices used in the algorithm, next and dist.
+     *
+     * @data (Object) Data structure used in main function.
+     * @edges (Array) List of edges where each is an Object contains a directed source and
+     *      target parameter.
+     * @returns Does not a return a value, operates directly on the data structure.
+     */
+    gpii.ontologyHandler.floydWarshall.algorithmInitMatrices = function (data, edges) {
+        // Create |V|x|V| dist and next arrays and initialize to Infinity and null
+        data.dist = fluid.generate(data.numVertices, function () {
+            return fluid.generate(data.numVertices, Infinity, false);
+        }, true);
+        data.next = fluid.generate(data.numVertices, function () {
+            return fluid.generate(data.numVertices, null, false);
+        }, true);
+        fluid.each(edges, function (edge) {
+            var u = data.vertices[edge.source],
+                v = data.vertices[edge.target];
+            data.dist[u][v] = 1; // All weights currently 1
+            data.next[u][v] = v;
+        });
+    };
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshall.fullAlgorithmWithReconstruction
+     * to perform the core algorithm of Floyd-Warshall.
+     *
+     * @dist (2D array) See algorithm documentation
+     * @next (2D array) See algorithm documentation
+     * @numVertices (Integer) total number of vertices
+     * @returns Does not a return a value, operates directly on the dist and next matrices.
+     */
+    gpii.ontologyHandler.floydWarshall.algorithmCore = function (dist, next, numVertices) {
+        // Standard Floyd-Warshall implementation
+        for (var k = 0; k < numVertices; k++) {
+            for (var i = 0; i < numVertices; i++) {
+                for (var j = 0; j < numVertices; j++) {
+                    if (dist[i][k] + dist[k][j] < dist[i][j]) {
+                        dist[i][j] = dist[i][k] + dist[k][j];
+                        next[i][j] = next[i][k];
+                    }
+                }
+            }
+        }
+    };
+
+    /*
+     * Uses the data generated in floydWarshall.fullAlgorithmWithReconstruction to
+     * generate the shortest path as an array of vertex names.
+     *
+     * @data (Object) - The data structure returned from
```

Comment:
Updated function docs for the entire file (even functions not part of this jira) 

We should probably open a global Jira to go through and update all the files for the project.